core/src/main/java/org/apache/accumulo/core/client/impl/MasterClient.java

```diff
@@ -44,14 +44,14 @@
   public static MasterClientService.Client getConnectionWithRetry(ClientContext context) {
     while (true) {
 
-      MasterClientService.Client result = getConnection(context);
+      MasterClientService.Client result = getConnection(context, false);
       if (result != null)
         return result;
       sleepUninterruptibly(250, TimeUnit.MILLISECONDS);
     }
   }
 
-  public static MasterClientService.Client getConnection(ClientContext context) {
+  public static MasterClientService.Client getConnection(ClientContext context, boolean isAdminRequest) {
```

Comment:
I'm confused about the `isAdminRequest`. The boolean flag should be named based on what it controls, not what kind of request the caller is going to use it for. In this case, maybe something like `boolean withTimeout`. Or better yet, pass the timeout in as an integer: `long timeoutMillis`, and then the `if` statement below becomes `if (timeoutMillis > 0)`

Reply:
Sorry but we agreed on a discussion that we really don't want to have that. The isAdminRequest will mean that it should not try forever as it is a administration task like shutdown. That's why we need to differ here. My Initial PR has done exactly that we conclude that what timeout should then we use  ? That's why we discard that option